### ISSUE TITLE
PK$ANNOTATION: formula_count

### ISSUE THREAD

**Issue by tsufz** [MassBank/MassBank-web#93]
Hi, what was the reason to introduce the formula count in the annotation? This actually not documented in the record format. I guess, this was introduced during RMassBank development? However, this makes problems with the design of the new database scheme. Many records don't have the formula count and I guess only the RMassBank generated ones have it, right?

If there is a good reason to show it, then we need to extend the record format. If we don't need it necessarily, I suggest to skip it in future and to remove it from existing records.

**Comment by meowcat** [comment]
Hi,

the formula_count is there to show how many formulas would have matched to a candidate peak. This is in order to give the user some confidence about the assigned formula; i.e. if there are multiple possible formulas, they should be more critical of a proposed assignment.

The MassBank record format 2.10 (section 2.6.2) specifies that the PK$ANNOTATION section shall be freely defined by the record author:

> Line 1 defines the record format of the annotation blocks. Contributors freely define 
> the record format by using appropriate terms

where Line 1 defines the table "columns" and the subsequent lines are the table rows. Or that's how I understood it. 

I don't really know how many (if any) non-RMassBank records use the feature PK$ANNOTATION, but this is how it was defined. I found it quite convenient to allow freeform definition of these columns; but if there is more standardization desired we should think carefully what to allow.

**Comment by schymane** [comment]
I agree with Michele, this information should be kept as it is essential.
The way it is currently defined, the annotation is coupled with the header field, which defines how to interpret the annotations.
There are other records with a different scheme and we will possibly look at different schemes again in the future.
Hopefully there's a way you can structure the database to account for and keep this flexibility, as it is useful.
________________________________________

**Comment by tsufz** [comment]
Well, then the DB scheme needs a more flexible design for this part. I agreed with @naperone that we will write issues for each problem showing up with record parsing and validation, including inconsistency with record format. Or inconsistency of record format itself (we found an example).

**Comment by meowcat** [comment]
But it is a legitimate question if some of the PK$ANNOTATION fields should be standardized, such that they can be stored in a database table directly (instead of a more complex [ID, row, column, value]-like format) or if we even want to fix an annotation scheme once and for all, I mean it's not an obvious choice but not completely unthinkable, right?

**Comment by schymane** [comment]
I agree we should standardise as far as possible.
A nice extension would be structures, which I started playing with in a few ET records...
But we could probably define a tight set of cases to be as parseable as possible...
________________________________________

**Comment by tsufz** [comment]
Wow, we are back to #81. I agree with @meowcat that we should avoid arbitrary inputs in order to avoid to store less usable data (because it is in a fancy format) and thus produce slack. All input should be well defined in the RF. We should avoid too much exceptional coding.
